charger/daheimladen.go

```diff
@@ -0,0 +1,317 @@
+	err = json.NewDecoder(res.Body).Decode(&remoteStopRes)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling response, err: %v", err)
+	}
+	if remoteStopRes.Status != "Accepted" {
+		return fmt.Errorf("charging station refused to stop transaction")
+	}
+	return nil
+}
+
+// MaxCurrent implements the api.Charger interface
+func (c *DaheimLaden) MaxCurrent(current int64) error {
+	changeConfigReq := daheimladen.ChangeConfigurationRequest{
+		Key:   "ChargeRate",
+		Value: fmt.Sprint(current),
+	}
+	data, err := json.Marshal(changeConfigReq)
+	if err != nil {
+		return fmt.Errorf("error marshaling request, err: %v", err)
+	}
+	req, err := http.NewRequest("POST", fmt.Sprintf("https://api.daheimladen.com/v1/cs/%s/change_config", c.chargingStationID), bytes.NewBuffer(data))
+	if err != nil {
+		return fmt.Errorf("error creating request, err: %v", err)
+	}
+	res, err := c.Do(req)
+	if err != nil {
+		return fmt.Errorf("error getting response, status-code: %v, err: %v", res.Status, err)
+	}
+	defer res.Body.Close()
+
+	changeConfigRes := daheimladen.ChangeConfigurationResponse{}
+	err = json.NewDecoder(res.Body).Decode(&changeConfigRes)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling response, err: %v", err)
+	}
+
+	if changeConfigRes.Status != "Accepted" {
+		return fmt.Errorf("error changing max current of charging station")
+	}
+	return c.MaxCurrentMillis(float64(current) * 1e3)
+}
+
+var _ api.ChargerEx = (*DaheimLaden)(nil)
+
+// MaxCurrentMillis implements the api.ChargerEx interface
+func (c *DaheimLaden) MaxCurrentMillis(current float64) error {
+	return nil
+}
+
+// Status implements the api.Charger interface
+func (c *DaheimLaden) Status() (api.ChargeStatus, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("https://api.daheimladen.com/v1/cs/%s/status", c.chargingStationID), nil)
+	if err != nil {
+		return api.StatusNone, fmt.Errorf("error creating request, err: %v", err)
+	}
+	res, err := c.Do(req)
+	if err != nil {
+		return api.StatusNone, fmt.Errorf("error getting response, status-code: %v, err: %v", res.Status, err)
+	}
+
+	status := daheimladen.GetLatestStatus{}
+	err = json.NewDecoder(res.Body).Decode(&status)
+	if err != nil {
+		return api.StatusNone, fmt.Errorf("error unmarshalling response, err: %v", err)
+	}
+
+	switch status.Status {
+	case string(daheimladen.AVAILABLE):
+		return api.StatusA, nil
+	case string(daheimladen.PREPARING):
+		return api.StatusB, nil
+	case string(daheimladen.CHARGING):
+		return api.StatusC, nil
+	case string(daheimladen.FAULTED):
+		return api.StatusF, nil
+	default:
+		return api.StatusNone, fmt.Errorf("none of the status matches with the status recieved from charging station, status: %s", status.Status)
+	}
+}
+
+var _ api.Meter = (*DaheimLaden)(nil)
+
+// CurrentPower implements the api.Meter interface
+func (c *DaheimLaden) CurrentPower() (float64, error) {
+	getConfigReq := daheimladen.GetConfigurationRequest{
+		Key: "ChargeRate",
```

Comment:
do you accept full Amps only?

Reply:
Yes at the moment our charging stations only accept integer values for amps